src/flash/nor/wcharm.c

```diff
@@ -510,9 +513,13 @@ static int ch32x_get_device_id(struct flash_bank *bank, uint32_t *device_id)
 static int ch32x_get_flash_size(struct flash_bank *bank, uint16_t *flash_size_in_kb)
 {	
 	struct target *target = bank->target;
-	uint32_t cpuid, flash_size_reg;
-    uint32_t temp;
-	int retval = target_read_u32(target, 0x1ffff7e0, flash_size_in_kb);	
+	// FIXME: this is almost certainly an actual bug and I'm applying pain killer
+	// to it.
+	// target_read_u32 reads 32 bits and stores it in the pointee of the final arg.
+	// The final argument here is a uint16_t as it is in the caller. I'm going to
+	// change this to read 16 bits, but this could cause synchronization issues.
+	// Orig: int retval = target_read_u32(target, 0x1ffff7e0, flash_size_in_kb);
+	int retval = target_read_u16(target, 0x1ffff7e0, flash_size_in_kb);
```

Comment:
Without debug docs from WCH, we don't know if this is wrong or right.  It might only allow a 32bit read, even if they are only interested in the lower 16bits.  Is the code gross? yes! is this code actually wrong? we have no idea!